writer.go

```diff
@@ -687,13 +698,18 @@ func (w *writer) run() {
 			if len(batch) == 0 {
 				continue
 			}
+			if conn != nil && time.Now().After(connIdleDeadline) {
```

Comment:
Should this appear right above the `if len(batch) == 0` ? https://github.com/segmentio/kafka-go/blob/150aa3726777853d906d78cc1a764093e7d597d5/writer.go#L698

That way even if there is no activity the connection gets properly closed when it reaches its deadline.